### PR TITLE
Fixes bot username override issue with incoming webhooks

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
In Global thread item list bot (incoming webhook) username override wasn’t implemented which this pr will fix

#### Ticket Link
Fixes [https://mattermost.atlassian.net/browse/MM-38180](https://mattermost.atlassian.net/browse/MM-38180)

#### Related Pull Requests
NONE

#### Screenshots
Screenshot 2021-09-08 at 10 17 24 PM

#### Release Note
```release-note
* Global thread item username (incoming webhook) username override fix
```

### CODE REVIEW

**Review by Willyfrog — APPROVED** [review]
LGTM

**Review by jgilliam17 — APPROVED** [review]
Thanks @AshishDhama 
Tested, looks good to merge.
- Verified incoming webhook username displays as expected, no longer displaying bot owner name.

**Comment by jgilliam17** [comment]
@AshishDhama I didn't merge since Kyriakos left a suggestion.

File: components/threading/global_threads/thread_item/thread_item.tsx [hunk 1]
```diff
@@ -152,7 +152,7 @@ function ThreadItem({
                         )}
                     </div>
                 )}
-                <span>{displayName}</span>
+                <span>{post.props?.override_username ? post.props.override_username : displayName}</span>
```
[the remark this answers is not in the document]
Comment (AshishDhama) [reply components/threading/global_threads/thread_item/thread_item.tsx]: Thanks,

**Comment by Willyfrog** [comment]
still LGTM :)

**Comment by AshishDhama** [comment]
@koox00 @jgilliam17 please take another look.

**Review by koox00 — APPROVED** [review]

**Pull request merged**